services/parser/parser.go

```diff
@@ -67,6 +71,23 @@ func GetInterval(value int, minInterval, maxInterval time.Duration) time.Duratio
 	return time.Duration(pMax)
 }
 
+func GetBlocksByRound(blockTime int) int64 {
+	switch {
+	case blockTime < 4000:
+		return 25
+	case blockTime >= 4000 && blockTime < 8000:
+		return 18
+	case blockTime >= 8000 && blockTime < 12000:
+		return 12
+	case blockTime >= 12000 && blockTime < 16000:
+		return 10
+	case blockTime >= 16000:
+		return 8
+	default:
+		return DefaultBlocksPerRound
```

Comment:
I suggest to remove `GetBlocksByRound` function, because you have control over how often you check for blocks instead. 

Two ways:
check blocks every 1 sec, get 5 blocks
check every 5 sec, get 25 blocks. 

This kinda solves the same problem. so you can control param #1, just define how often you check.

Ideally there is a limit of 10 blocks per blockchain at the time. It could be *2 if behind as an example. 

But `GetBlocksByRound` implementation is overkill as well